### DISCUSSION TITLE
mail.err - "Execution of script /usr/lib/dovecot/sieve-pipe/learn-ham.sieve failed"

### DISCUSSION THREAD

**Discussion by blakegonzales** [docker-mailserver/docker-mailserver#3604] [General]
When attempting to train rpamd by moving an email into or out of `Junk` I get the following errors in `mail.err`

```
Oct 28 04:44:47 mail dovecot: imap(xxxx@xxxxx.com)<201906><1WsY18MIQsJFBOpq>: Error: sieve: failed to execute to program `rspamc': refer to server log for more information. [2023-10-28 04:44:47]
Oct 28 04:44:47 mail dovecot: imap(xxxx@xxxxx.com)<201906><1WsY18MIQsJFBOpq>: Error: sieve: Execution of script /usr/lib/dovecot/sieve-pipe/learn-spam.sieve failed
Oct 28 04:44:55 mail dovecot: imap(xxxx@xxxxx.com)<201906><1WsY18MIQsJFBOpq>: Error: sieve: failed to execute to program `rspamc': refer to server log for more information. [2023-10-28 04:44:55]
Oct 28 04:44:55 mail dovecot: imap(xxxx@xxxxx.com)<201906><1WsY18MIQsJFBOpq>: Error: sieve: Execution of script /usr/lib/dovecot/sieve-pipe/learn-ham.sieve failed
```
I'm on release v12.1.0.

How can I fix this and determine if the training is working?

**Comment by polarathene** [comment]
> How can I fix this and determine if the training is working?

Try with `:edge` to verify if it's been fixed. Otherwise you would need to wait until `v13`, which is likely to be released in November, or at least by December AFAIK.

If you still encounter the issue with `:edge` / `v13`, please open a bug report.

**Comment by blakegonzales** [comment]
`:edge` didn't fix it.

I'll wait until v13.  Thank you.

**Comment by Nebucatnetzer** [comment]
I see the same on 13.0.1

**Comment by georglauterbach** [comment]
I'm running `:edge` and I am not seeing this issue. Did you properly restart with `docker compose down` and `docker compose up` (or the equivalent in other CRIs)? If you continue seeing this, please open a bug report :)

**Comment by Nebucatnetzer** [comment]
I run it directly without compose. I will open an issue tomorrow.On 30 Dec 2023, at 21:55, Georg Lauterbach ***@***.***> wrote:﻿
I'm running :edge and I am not seeing this issue. Did you properly restart with docker compose down and docker compose up (or the equivalent in other CRIs)? If you continue seeing this, please open a bug report :)

**Comment by georglauterbach** [comment]
Please make sure to attach the Rspme log :)

**Comment by Nebucatnetzer** [comment]
For me this was fixed with 13.1. on aarch64.
